src/katsdpcontroller/generator.py

```diff
@@ -1089,6 +1075,20 @@ def get_baseline_index(a1, a2):
                     default=stream.n_baselines,
                     initial_status=Sensor.Status.NOMINAL,
                 ),
+                Sensor(
+                    int,
+                    f"{stream.name}.n-chans",
+                    "The number of frequency channels in an integration",
+                    default=stream.n_chans,
+                    initial_status=Sensor.Status.NOMINAL,
+                ),
+                Sensor(
+                    int,
+                    f"{stream.name}.n-chans-per-substream",
+                    "Number of channels in each substream for this X-engine stream",
+                    default=stream.n_chans_per_substream,
+                    initial_status=Sensor.Status.NOMINAL,
+                ),
```

Comment:
I think you've misinterpreted a comment I made about the descriptions. These sensors still need to exist for tied-array-channelised-voltage streams as well. The descriptions just need to be reworded to be be generic.

Reply:
Too right, I went on a 50/50 that I should create them for the specific type of stream (because the ICD has [slightly] different descriptions for each type). 
* For the record, I did make a note that I would need to add this sensor for B-engine streams under NGC-447. 

Would you be happy with a description of
> Number of channels in each substream for this data stream

?